File: Analise-Jesse/Python/SplayTreeTest.py
import time
import random
import csv
import sys

sys.setrecursionlimit(600000)
from SplayTree import SplayTree

    #Aluno: Jessé Oliveira das Chagas
    #Email: jesse.oliveira.chagas@ccc.ufcg.edu.br
    #Matrícula: 123210473
    #Nome de usuário do GitHub: jessechagas

def gerar_dados_aleatorios(tamanho):
    return [random.randint(1, tamanho * 10) for _ in range(tamanho)]

def gerar_dados_ordenados(tamanho):
    return list(range(1, tamanho + 1))

def testar_splay_tree(valores):
    arvore = SplayTree()

    # Teste de inserção
    inicio = time.time()
    for v in valores:
        arvore.insert(v)
    tempo_insercao = time.time() - inicio

    # Teste de busca
    inicio = time.time()
    for v in valores:
        arvore.search(v)
    tempo_busca = time.time() - inicio

    # Teste de remoção
    inicio = time.time()
    for v in valores:
        arvore.delete(v)
    tempo_remocao = time.time() - inicio

    return tempo_insercao, tempo_busca, tempo_remocao

def executar_teste(entradas):
    print("Escolha o tipo de dados para o teste:")
    print("1 - Aleatórios")
    print("2 - Ordenados")
    escolha = input("Sua escolha (1 ou 2): ")

    tipo = "aleatorio" if escolha == "1" else "ordenado"
    file_name = f"dados_{tipo}_splay_python.csv"

    with open(file_name, mode="w", newline='') as arquivo_csv:
        writer = csv.writer(arquivo_csv)
        writer.writerow(["Tamanho", "Arvore", "Operacao", "Tempo"])

        for tamanho in entradas:
            print(f"\nExecutando testes para tamanho: {tamanho}")
            if escolha == "1":
                dados = gerar_dados_aleatorios(tamanho)
            else:
                dados = gerar_dados_ordenados(tamanho)

            tempos = testar_splay_tree(dados)

            writer.writerow([tamanho, "Splay", "Insercao", f"{tempos[0]:.6f}"])
            writer.writerow([tamanho, "Splay", "Busca", f"{tempos[1]:.6f}"])
            writer.writerow([tamanho, "Splay", "Remocao", f"{tempos[2]:.6f}"])

            print(f"Splay Tree - Inserção: {tempos[0]:.6f}s, Busca: {tempos[1]:.6f}s, Remoção: {tempos[2]:.6f}s")
            input("Pressione Enter para continuar...")

    print(f"\nDados salvos em '{file_name}'.")

if __name__ == "__main__":
    entradas = [10000, 50000, 100000, 250000, 500000, 1000000]
    executar_teste(entradas)

File: Analise-Jesse/Java/splayTree/SplayTreeTest.java
package splayTree;

import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;
import java.util.Scanner;

/*
    Aluno: Emerson Henrique Sulpino de Araújo
    Email: emerson.henrique.sulpino.araujo@ccc.ufcg.edu.br
    Matrícula: 123210141
    User do GitHub: Emerson349
*/

public class SplayTreeTest {

    // Gera dados aleatórios
    private int[] gerarDadosAleatorios(int tamanho) {
        Random random = new Random();
        int[] dados = new int[tamanho];
        for (int i = 0; i < tamanho; i++) {
            dados[i] = random.nextInt(tamanho * 10) + 1;
        }
        return dados;
    }

    // Gera dados ordenados
    private int[] gerarDadosOrdenados(int tamanho) {
        int[] dados = new int[tamanho];
        for (int i = 0; i < tamanho; i++) {
            dados[i] = i + 1;
        }
        return dados;
    }

    // Testa a SplayTree com inserção, busca e remoção
    private double[] testSplayTree(int[] valores) {
        SplayTree arvore = new SplayTree();
        long inicio;

        // Inserção
        inicio = System.nanoTime();
        for (int valor : valores) {
            arvore.insert(valor);
        }
        double tempoInsercao = (System.nanoTime() - inicio) / 1e9;

        // Busca
        inicio = System.nanoTime();
        for (int valor : valores) {
            arvore.search(valor);
        }
        double tempoBusca = (System.nanoTime() - inicio) / 1e9;

        // Remoção
        inicio = System.nanoTime();
        for (int valor : valores) {
            arvore.remove(valor);
        }
        double tempoRemocao = (System.nanoTime() - inicio) / 1e9;

        return new double[]{tempoInsercao, tempoBusca, tempoRemocao};
    }

    // Execução dos testes com entrada do usuário
    private void executarTeste(int[] tamanhos) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Escolha o tipo de dados para o teste:");
        System.out.println("1 - Aleatórios");
        System.out.println("2 - Ordenados");
        int escolha = scanner.nextInt();

        String tipo = (escolha == 1) ? "aleatorio" : "ordenado";
        String nomeArquivo = "dados_" + tipo + "_splay_java.csv";
        StringBuilder csvData = new StringBuilder("Tamanho,Arvore,Operacao,Tempo\n");

        for (int tamanho : tamanhos) {
            int[] dados = (escolha == 1) ? gerarDadosAleatorios(tamanho) : gerarDadosOrdenados(tamanho);

            double[] tempos = testSplayTree(dados);
            csvData.append(tamanho).append(",Splay,Inserção,").append(tempos[0]).append("\n");
            csvData.append(tamanho).append(",Splay,Busca,").append(tempos[1]).append("\n");
            csvData.append(tamanho).append(",Splay,Remoção,").append(tempos[2]).append("\n");

            System.out.println("Tamanho: " + tamanho);
            System.out.printf("Splay Tree - Inserção: %.6fs, Busca: %.6fs, Remoção: %.6fs%n",
                    tempos[0], tempos[1], tempos[2]);

            System.out.println("Pressione Enter para continuar...");
            scanner.nextLine(); // espera Enter
            scanner.nextLine(); // limpa buffer
        }

        try (FileWriter writer = new FileWriter(nomeArquivo)) {
            writer.write(csvData.toString());
            System.out.println("Dados salvos em '" + nomeArquivo + "'.");
        } catch (IOException e) {
            System.err.println("Erro ao salvar o CSV: " + e.getMessage());
        }
    }

    public static void main(String[] args) {
        SplayTreeTest teste = new SplayTreeTest();
        int[] entradas = {10000, 50000, 100000, 250000, 500000, 1000000};
        teste.executarTeste(entradas);
    }
}
